lib/ui/compositing/scene.cc

```diff
@@ -102,6 +100,42 @@ Dart_Handle Scene::toImage(uint32_t width,
   return Picture::RasterizeToImage(picture, width, height, raw_image_callback);
 }
 
+void Scene::RasterizeToImageSync(uint32_t width,
+                                 uint32_t height,
+                                 Dart_Handle raw_image_handle) {
+  auto* dart_state = UIDartState::Current();
+  auto unref_queue = dart_state->GetSkiaUnrefQueue();
+  auto snapshot_delegate = dart_state->GetSnapshotDelegate();
+  auto raster_task_runner = dart_state->GetTaskRunners().GetRasterTaskRunner();
+
+  auto image = CanvasImage::Create();
+  auto dl_image = DlDeferredImageGPU::Make(SkISize::Make(width, height));
+  image->set_image(dl_image);
+
+  auto layer_tree = takeLayerTree();
+
+  fml::TaskRunner::RunNowOrPostTask(
+      raster_task_runner,
+      [snapshot_delegate, unref_queue, dl_image = std::move(dl_image),
+       layer_tree = std::move(layer_tree.get()), width, height]() {
```

Comment:
What I'm doing to layer_tree doesn't seem right, but it did compile....

Reply:
I don't think we can safely call `takeLayerTree` here. That makes the scene unusable in the future.

That said, I think we'll have to make the layer_tree_ a shared_ptr for this to work to share it across threads.

Reply:
Also you don't need to `move` a raw pointer. 

You'll want something like `layer_tree = layer_tree_`, and make the ivar a `std::shared_ptr`.